program-runtime/src/loaded_programs.rs

```diff
@@ -102,6 +102,32 @@ impl From<ProgramCacheEntryOwner> for Pubkey {
     }
 }
 
+/*
+    The possible ProgramCacheEntryType transitions:
+    DelayVisibility is special because in the cache it is stored as Loaded and
+    is only materialized temporarily by ProgramCacheForTxBatch::find().
+
+    Builtin re/deployment:
+    - Empty => Builtin in TransactionBatchProcessor::add_builtin
+    - Builtin => Builtin in TransactionBatchProcessor::add_builtin
+
+    Un/re/deployment (with delay and cooldown):
+    - Empty / Closed => DelayVisibility in UpgradeableLoaderInstruction::DeployWithMaxDataLen
```

Comment:
Isn't the transition here => Loaded? DelayVisibility is only (temporarily) returned

Reply:
Yes it is, previously I described this state as `Loaded+DelayVisibility`, but that is confusing too.